packages/monitor-v2/src/monitor-og/SnapshotVerification.ts

```diff
@@ -104,27 +111,59 @@ const getGraphqlData = async (ipfsHash: string, url: string): Promise<GraphqlDat
     }
   `);
   try {
-    return await request<GraphqlData, { ipfsHash: string }>(url, query, { ipfsHash });
-  } catch (error) {
-    return new Error(JSON.stringify(error));
+    return await retry(
+      async () => {
+        return await request<GraphqlData, { ipfsHash: string }>(url, query, { ipfsHash });
+      },
+      {
+        ...retryOptions,
+        onRetry(error) {
+          // We only log the error from the last retry if all attempts failed.
+          lastTryError = error;
+        },
```

Comment:
Isn't the error thrown by `retry`, itself, the last error thrown by the inner function?

Put differently, couldn't we just catch it directly, instead of saving it here and returning it?

Reply:
You're right, this can be simplified by catching error directly